### PR TITLE
Azure docs

### LINKED ISSUE #444: Test kingfisher-main on Ubuntu on Azure

### PR DESCRIPTION
**Pull request by dogsbody-ashley**
closes #444

### CODE REVIEW

File: docs/deploy/create_server.rst [hunk 1]
```diff
@@ -104,6 +104,60 @@ Hetzner
...
+   #. Set *Username* to ocpadmin
+   #. Set *Root Password* to a `strong password <https://www.lastpass.com/features/password-generator>`__
+
+   #. Select a *Next: Disks >*
+   #. Set *OS Disk Type* to *Standard SSD* 
+      #. Create and attach an additional disk if required with the *Size* of *Standard SSD*
+
+   #. Select a *Next: Networking >*
+   #. Set *Virtual Network* to *SensibleName-vnet (e.g. kingfisher-vnet)*   
+   #. Set *Subnet* to use an IP range that does not overlap an already existing *Virtual Machine* in the same *Resource Group*
```
Comment (jpmckinney) [inline docs/deploy/create_server.rst]: In what scenario would two virtual machines be in the same resource group?

How is this done?

File: docs/deploy/create_server.rst [hunk 2]
```diff
@@ -104,6 +104,60 @@ Hetzner
...
+      #. Delete all default rules
+      #. *+ Add an inbound rule* for each individual item below:
+		Allow-SSH - Port 22 - Protocol TCP - Source Any - Priority 1030
+		Allow-HTTP - Port 80 - Protocol TCP - Source Any - Priority 1040
+		Allow-HTTPS - Port 443 - Protocol TCP - Source Any - Priority 1050 
+		Allow-ICMP - Port Any - Protocol ICMP - Source Any - Priority 1060 
+		Allow-Prometheus - Port 7231 - Protocol TCP - Source 139.162.253.17/32 - Priority 1080 
+		Allow-Prometheus-IPv6 - Port 7231 - Protocol TCP - Source 2a01:7e00::f03c:93ff:fe13:a12c/128 - Priority 1100
+      .. note::
+      If using Docker, configure an external firewall - https://ocdsdeploy.readthedocs.io/en/latest/develop/update/firewall.html#linode-firewall)
```
Comment (jpmckinney) [inline docs/deploy/create_server.rst]: What options does Azure have for external firewalls?

**Review by jpmckinney — COMMENTED** [review]
You'll need to pull my changes first.

**Comment by dogsbody-ashley** [comment]
Hi James,

Thank you for the comments, I'll get the answers and revise.
Regarding Reverse DNS, yes there will be a process, I just haven't set this up in your Azure account yet.

File: docs/deploy/create_server.rst (hunk 1 above)
Comment (dogsbody-ashley) [reply docs/deploy/create_server.rst]: I'll removed this. 
If you had a separate DB/Web server situation, you could place everything around the servers into the same resource group.

File: docs/deploy/create_server.rst (hunk 2 above)
Comment (dogsbody-ashley) [reply docs/deploy/create_server.rst]: I've updated this for docker and non docker.

**Comment by jpmckinney** [comment]
Thank you for the additional commits @dogsbody-ashley

Let me know if it is ready for review.

File: docs/deploy/create_server.rst (hunk 1 above)
Comment (jpmckinney) [reply docs/deploy/create_server.rst]: Bytemark also offered an option to group VMs, but it just became unnecessary overhead. Is there any consequence to just putting everything in a "default" group?

File: docs/deploy/create_server.rst [hunk 3]
```diff
@@ -202,6 +202,115 @@ Create the server via the :ref:`host<hosting>`'s interface.
...
+            #. Click the *Change size* link
+            #. Select "Standard SSD" from the *Storage type* dropdown
+            #. Click the desired size
+            #. Click the *OK* button
+
+      #. Click the *Next : Networking >* button
+
+         #. Set *Virtual network* to an appropriate name with a ``-vnet`` suffix (e.g. ``kingfisher-vnet``)
+         #. Set *Subnet* to *default (10.0.0.0/24)*
+         #. Set *Public IP* to the server's FQDN (e.g. ``ocp42.open-contracting.org``)
```
Comment (jpmckinney) [inline docs/deploy/create_server.rst:238]: This opens a sidebar with a choice of "Routing preference". Is the default ("Microsoft network") the appropriate option?

Screenshot 2023-11-07 at 4 04 03 PM

File: docs/deploy/create_server.rst [hunk 4]
```diff
@@ -202,6 +202,115 @@ Create the server via the :ref:`host<hosting>`'s interface.
...
+
+         #. If not using Docker, set *NIC network security group* to *None*.
+
+         #. If using Docker, set *NIC network security group* to *Advanced*
+
+            #. Click *Create new*
```
Comment (jpmckinney) [inline docs/deploy/create_server.rst:244]: Is it necessary to create a new one, or can the same network security group be reused? (e.g. we can define a "web" group that allows 22, 80, 443, ICMP and Prometheus, like below)

Comment (dogsbody) [reply docs/deploy/create_server.rst]: You aren't using generic network security groups elsewhere so we have followed your existing processes.

These are network security group's so they are already reused when a subnet is reused (they are per subnet, not per instance). 

If a server already exists inside the same subnet then you can reuse the same network security group for this subnet.

**Pull request merged**